src/Marian/index.ts

```diff
@@ -60,10 +60,15 @@ export default class Marian {
     if (!url) {
       assert.fail('Assertion: Missing url');
     }
-    const parsedUrl = new URL(url, `http://${req.headers.host}`);
-
+    let parsedUrl: URL;
+    try {
+      parsedUrl = new URL(url, `http://${req.headers?.host}`);
+    } catch (e) {
+      console.error(`URL constructor could not create a URL with url ${url} and base ${req.headers?.host}`);
+      console.trace();
+      throw e;
```

Comment:
I was thinking about this error, and I'm wondering if we should just avoid logging an error (or maybe just log a warning) and send an error as a response, instead of throwing.

Otherwise, I'm afraid we'll keep logging noisy errors for parsing urls from requests we can't control. Any thoughts?

Reply:
Sending an error as a response instead makes a lot of sense. For the purpose of investigating the erroring requests right now, I think  it would be helpful to also log a warning with information on bad requests